GDevelop.js/__tests__/GDJSAsyncCodeGenerationIntegrationTests.js

```diff
@@ -3568,4 +3571,77 @@ describe('libGD.js - GDJS Async Code Generation integration tests', function ()
       );
     });
   });
+
+  describe('Events based asynchronous functions', () => {
+    it('Resolves after awaiting an async action', () => {
+      const { gdjs, runtimeScene } = makeMinimalGDJSMock();
+      const extensionModule =
+        generateCompiledEventsForSerializedEventsBasedExtension(
+          gd,
+          require('./extensions/EBAsyncAction.json'),
+          gdjs,
+          runtimeScene
+        );
+
+      const {
+        free: { wait },
+        behaviors: { Behavior },
+        objects: { Object },
+      } = extensionModule;
+
+      const freeExtensionTask = wait();
+      expect(freeExtensionTask.update()).toBe(false);
+      expect(runtimeScene.getAsyncTasksManager().tasks.size).toBe(1);
+      runtimeScene.getAsyncTasksManager().markAllFakeAsyncTasksAsFinished();
+      runtimeScene.getAsyncTasksManager().processTasks();
+      expect(freeExtensionTask.update()).toBe(true);
+
+      const obj = new Object(runtimeScene, {});
+      const objectExtensionTask = obj.wait();
+      expect(objectExtensionTask.update()).toBe(false);
+      expect(runtimeScene.getAsyncTasksManager().tasks.size).toBe(1);
+      runtimeScene.getAsyncTasksManager().markAllFakeAsyncTasksAsFinished();
+      runtimeScene.getAsyncTasksManager().processTasks();
+      expect(objectExtensionTask.update()).toBe(true);
```

Comment:
Ubernit: avoid abbreviations.

```suggestion
      const object = new Object(runtimeScene, {});
      const objectExtensionTask = object.wait();
      expect(objectExtensionTask.update()).toBe(false);
      expect(runtimeScene.getAsyncTasksManager().tasks.size).toBe(1);
      runtimeScene.getAsyncTasksManager().markAllFakeAsyncTasksAsFinished();
      runtimeScene.getAsyncTasksManager().processTasks();
      expect(objectExtensionTask.update()).toBe(true);
```

Reply:
For some reason I thought `object` was a reserved keyword 🤦‍♂️